### PR TITLE
Go SDK Tweaks

### PR DESCRIPTION
**Pull request by aludvik**

### CODE REVIEW

File: sdk/go/src/sawtooth_sdk/client/encoder.go [hunk 1]
```diff
@@ -243,6 +246,12 @@ func Decode(encoded, encoding string) []byte {
 			panic("Couldn't decode")
 		}
 		return bytes
+	case "base64":
+		bytes, err := base64.StdEncoding.DecodeString(encoded)
+		if err != nil {
+			panic("Couldn't decode")
```
Comment (vaporos) [inline sdk/go/src/sawtooth_sdk/client/encoder.go]: This Decode() func seems to do the same thing as base64.StdEncoding.DecodeString() only panic instead of returning an error.  Seems like we should instead just use the libraries directly so clients can handle the errors.

File: sdk/go/src/sawtooth_sdk/messaging/connection.go [hunk 2]
```diff
@@ -76,7 +76,6 @@ func NewConnection(context *zmq.Context, t zmq.Type, uri string) (*Connection, e
 
 	identity := GenerateId()
 	socket.SetIdentity(identity)
-	logger.Debug("Socket Identity set to ", identity)
```
Comment (ojalatodd) [inline sdk/go/src/sawtooth_sdk/messaging/connection.go]: General question: Why did you remove the Debug level logger?

File: sdk/go/src/sawtooth_sdk/client/encoder.go (hunk 1 above)
Comment (aludvik) [inline sdk/go/src/sawtooth_sdk/client/encoder.go]: Good point, I'll change it.

File: sdk/go/src/sawtooth_sdk/messaging/connection.go (hunk 2 above)
Comment (aludvik) [inline sdk/go/src/sawtooth_sdk/messaging/connection.go]: For anyone using the SDK, it generates a lot of logging output unrelated to what they are trying to use it for. The alternative would be to put more work into the sdk/logging package so that log levels can be set per-package or per-file.

**Comment by aludvik** [comment]
Dropped the base64 commit and rename commit and added a commit to change the encode/decode methods. See commit message for rationale.

File: sdk/go/src/sawtooth_sdk/client/encoder.go [hunk 3]
```diff
@@ -224,28 +224,20 @@ func ParseBatches(b []byte) ([]*Batch, error) {
...
-// Encode converts the given byte slice to a string using the given encoding.
-func Encode(key []byte, encoding string) string {
-	switch encoding {
-	case "hex":
-		return strings.ToLower(hex.EncodeToString(key))
-	default:
-		panic("Unsupported encoding: " + encoding)
-	}
+// MustEncode converts the given byte slice to a lower-case hex-encoded string.
+// Panics if encoding failes.
```
Comment (agunde406) [inline sdk/go/src/sawtooth_sdk/client/encoder.go]: failes -> fails
